### PR TITLE
Update to smallrye-jwt-2.4.0

### PR DESCRIPTION
**Pull request by sberyozkin**

### CODE REVIEW

File: docs/src/main/asciidoc/security-jwt.adoc [hunk 1]
```diff
@@ -807,6 +807,19 @@ Finally both the confidentiality and integrity of the claims can be further enfo
...
 SmallRye JWT provides an API for securing the JWT claims using all of these options.
 
+== Maven dependency
+
+[source,xml]
+----
+<dependency>
+  <groupId>io.smallrye</groupId>
+  <artifactId>smallrye-jwt-build</artifactId>
+  <version>${smallrye.jwt.version}</version>
```
Comment (gastaldi) [inline docs/src/main/asciidoc/security-jwt.adoc:817]: Is the version still necessary here?

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-jwt.adoc:817]: @gastaldi Sorry, missed this question - I don't know - it is not in Quarkus `bom/application/pom.xml` dependencies yet as at the moment `quarkus-smallrye-jwt`/`quarkus-oidc` depend on it indirectly.
I can update this section later, as part of the fix for #13499, to include `io.quarkus/quarkus-smallrye-jwt-build` (no version) ?

Comment (gsmet) [reply docs/src/main/asciidoc/security-jwt.adoc:817]: I think adding it to the BOM should be done in the same commit.

**Review by gsmet — CHANGES_REQUESTED** [review]

**Comment by sberyozkin** [comment]
@gastaldi @gsmet, ok, np, I suppose even at this stage it would be easier for the users to have it in the BOM

File: docs/src/main/asciidoc/security-jwt.adoc (hunk 1 above)
Comment (gsmet) [reply docs/src/main/asciidoc/security-jwt.adoc:817]: So you can remove the version now from here now as it will automatically come with the BOM.

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-jwt.adoc]: @gsmet, yeah, sorry, I realized it once I switched my laptop off and went for a walk :-), sorry :-)

**Review by gsmet — APPROVED** [review]

**Pull request merged**